src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerInitializationHostedService.cs

```diff
@@ -51,8 +52,15 @@ private async Task ApplyStartContextIfPresent(CancellationToken cancellationToke
                 var assignmentContext = _startupContextProvider.SetContext(encryptedAssignmentContext);
                 await SpecializeMSISideCar(assignmentContext);
 
-                bool success = _instanceManager.StartAssignment(assignmentContext);
-                _logger.LogDebug($"StartAssignment invoked (Success={success})");
+                try
+                {
+                    bool success = await _instanceManager.AssignInstanceAsync(assignmentContext);
+                    _logger.LogDebug("AssignInstanceAsync was invoked (Success={success})", success);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Failed to assign instance.");
+                }
```

Comment:
Added tests for this in `AtlasContainerInitializationHostedServiceTests` and `LegionContainerInitializationHostedServiceTests` - previous fire and forget pattern didn't observe exceptions related to assignment failure.